### DISCUSSION TITLE
How do I change format?

### DISCUSSION THREAD

**Discussion by ghost** [openai/whisper#1111] [Q&A]
I want output_format to be text and srt. Tried different commands for cmd and editing a file in the python folder to no avail. I'm a noob, please help.

**Comment by glangford** [comment]
**From the command line**

You can specify an individual format (one of "txt", "vtt", "srt", "tsv", "json") using `--output_format`. But since you want two formats, you would need to specify `--output_format all` then delete the outputs you don't need.

**In Python**

`whisper` writes output like this
```python
writer = get_writer(output_format, output_dir)
writer(result, audio_path)
```

So if you are comfortable in Python, to create just txt and srt you can do something like this:
```python
from whisper.utils import get_writer

result = model.transcribe(the_file, ... )
srt_writer = get_writer("srt", path)
srt_writer(result, path)
txt_writer = get_writer("txt", path)
txt_writer(result, path)
```

**Reply by ard-av** [reply]
Greetings.
02.2023 so it worked. Now it gives an error. Please help correct the code
![Screenshot_1](https://user-images.githubusercontent.com/114281218/235297564-597369c7-227d-40ac-8318-41a41c18c0f0.jpg)
![Screenshot_2](https://user-images.githubusercontent.com/114281218/235297567-bf9397ef-eefd-4a2c-9ed5-710aaf49ba7a.jpg)

**Reply by Pareek-Yash** [reply]
did you find the solution?

**Reply by glangford** [reply]
@ard-av @Pareek-Yash Try
```python
    srt_writer(result, file_name, {"max_line_width":50, "max_line_count":2, "highlight_words":False} ) # SRT: dictionary for 'options' 
```

**Comment by ghost** [comment]


**Reply by glangford** [reply]
You can only pick one of the strings listed in `choices`. I suggest you use `all` and delete the output you don't need.

**Reply by jongwook** [reply]
There's #1039 that adds support for something like `--output_format txt,srt`, I should probably test/polish it a bit and merge that

**Reply by mikkovedru** [reply]
@jongwook 
That would be really nice! ;-)
